src/ORBextractor.cc

```diff
@@ -895,6 +949,97 @@ namespace ORB_SLAM3
+
+        for (int level = 0; level < nlevels; ++level)
+        {
+            const int minBorderX = EDGE_THRESHOLD-3;
+            const int minBorderY = minBorderX;
+            const int maxBorderX = mvImagePyramid[level].cols-EDGE_THRESHOLD+3;
+            const int maxBorderY = mvImagePyramid[level].rows-EDGE_THRESHOLD+3;
+
+            const float width = (maxBorderX-minBorderX);
+            const float height = (maxBorderY-minBorderY);
+
+            const int nCols = width/W;
+            const int nRows = height/W;
+            const int wCell = ceil(width/nCols);
+            const int hCell = ceil(height/nRows);
+
+
+            vector<cv::KeyPoint> vToDistributeKeys = vToDistributeKeysCalculate(nRows, minBorderY,
+                                                                                hCell, maxBorderY,
+                                                                                nCols, minBorderX,
+                                                                                wCell, maxBorderX,
+                                                                                level, false);
+            vector<cv::KeyPoint> vToDistributeKeysS = vToDistributeKeysCalculate(nRows, minBorderY,
+                                                                                hCell, maxBorderY,
+                                                                                nCols, minBorderX,
+                                                                                wCell, maxBorderX,
+                                                                                level, true);
+
+            vector<KeyPoint> & keypoints = allKeypoints[level];
+            keypoints.reserve(nfeatures);
+
+            vector<KeyPoint> keypointsTemp = DistributeOctTree(vToDistributeKeys, minBorderX, maxBorderX,
+                                          minBorderY, maxBorderY,mnFeaturesPerLevel[level], level);
+            vector<KeyPoint> keypointsSTemp = DistributeOctTree(vToDistributeKeysS, minBorderX, maxBorderX,
+                                                               minBorderY, maxBorderY,mnFeaturesPerLevel[level], level);
+
+            float x, y, z, u, v;
+            vector<cv::KeyPoint> newKeypointsSTemp;
+            for(auto & i : keypointsSTemp)
+            {
+                z = depthS.at<float>((int)i.pt.y, (int)i.pt.x);
+                x = (i.pt.x - KS.at<float>(0, 2)) * z / KS.at<float>(0, 0);
+                y = (i.pt.y - KS.at<float>(1, 2)) * z / KS.at<float>(1, 1);
+
+                Eigen::Vector4f xyz(x, y, z, 1);
+                Eigen::Vector4f newXyz = T * xyz;
+                x = newXyz[0];
+                y = newXyz[1];
+                z = newXyz[2];
+
+                u = round(((K.at<float>(0, 0) * x) / z) + K.at<float>(0, 2));
+                v = round(((K.at<float>(1, 1) * y) / z) + K.at<float>(1, 2));
+
+                if ((u > 0) and (u < depthS.cols) and (v > 0) and (v < depthS.rows)){
```

Comment:
Looks like we don't need those checks, in such way we cut all keypoints outside master image